src/test/java/Hardcore/page/GoogleCloudPricingCalculatorPage.java

```diff
@@ -0,0 +1,142 @@
+import org.openqa.selenium.support.FindBy;
+
+public class GoogleCloudPricingCalculatorPage extends AbstractPage {
+
+    @FindBy(xpath = "//iframe[contains(@src, '/products/calculator')]")
+    WebElement frameMain;
+
+    @FindBy(xpath = "//md-tab-item[@aria-controls='tab-content-1']")
+    WebElement buttonComputeEngine;
+
+    @FindBy(xpath = "//input[@ng-model='listingCtrl.computeServer.quantity']")
+    WebElement inputNumberOfInstances;
+
+    @FindBy(xpath = "//input[@ng-model='listingCtrl.computeServer.label']")
+    WebElement inputPurpose;
+
+    @FindBy(xpath = "//md-select[@ng-model='listingCtrl.computeServer.os']")
+    WebElement dropdownSoftware;
+    @FindBy(xpath = "//md-option[@value='free']")
+    WebElement itemSoftware;
+
+    @FindBy(xpath = "//md-select[@ng-model='listingCtrl.computeServer.class']")
+    WebElement dropdownClass;
+    @FindBy(xpath = "//md-select-menu[@style]/descendant::md-option[@value='regular']")
+    WebElement itemClass;
+
+    @FindBy(xpath = "//md-select[@name='series']")
+    WebElement dropdownSeries;
+    @FindBy(xpath = "//md-option[@value='n1']")
+    WebElement itemSeries;
+
+    @FindBy(xpath = "//md-select[@placeholder='Instance type']")
+    WebElement dropdownMachineType;
+    @FindBy(xpath = "//md-option[@value='CP-COMPUTEENGINE-VMIMAGE-N1-STANDARD-8']")
+    WebElement itemMachineType;
+
+    @FindBy(xpath = "//md-checkbox[@aria-label='Add GPUs' and @ng-model='listingCtrl.computeServer.addGPUs']")
+    WebElement checkboxAddGPUs;
+    @FindBy(xpath = "//md-select[@placeholder='Number of GPUs']")
+    WebElement dropboxNumberOfGPUs;
+    @FindBy(xpath = "//md-option[@value='1' and contains(@ng-repeat, 'gpuType')]")
+    WebElement itemNumberOfGPUs;
+    @FindBy(xpath = "//md-select[@placeholder='GPU type']")
+    WebElement dropboxGPUType;
+    @FindBy(xpath = "//md-option[@value='NVIDIA_TESLA_V100']")
+    WebElement itemGPUType;
+
+    @FindBy(xpath = "//md-select[@placeholder='Local SSD']")
+    WebElement dropdownSSD;
+    @FindBy(xpath = "//md-option[@value='2' and contains(@ng-repeat, 'item in listingCtrl.dynamicSsd.computeServer')]")
+    WebElement itemSSD;
+
+    @FindBy(xpath = "//md-select[@placeholder='Datacenter location'][@ng-model='listingCtrl.computeServer.location']")
+    WebElement dropdownLocation;
+    @FindBy(xpath = "//md-select-menu[@class='md-overflow']/descendant::md-option[@value='europe-west3' and contains(@ng-repeat, 'item in listingCtrl.fullRegionList')]")
+    WebElement itemLocation;
+
+    @FindBy(xpath = "//md-select[@placeholder='Committed usage']")
+    WebElement dropdownUsage;
+    @FindBy(xpath = "//div[@class='md-select-menu-container md-active md-clickable']/descendant::md-option[@ng-value='1'][@value='1']")
+    WebElement itemUsage;
+
+    @FindBy(xpath = "//button[@aria-label='Add to Estimate' and contains(@ng-click, 'addComputeServer')]")
+    WebElement buttonAddToEstimate;
+
+    @FindBy(xpath = "//h2/b[@class='ng-binding']")
```

Comment:
Good to know, thanks. It was done like that in the course's videos, so I use it too. On top of that, I tried to use unique locators everywhere, so that there is only one single occurrence of it on every page.